crates/sp-executor/src/lib.rs

```diff
@@ -440,6 +440,14 @@ sp_api::decl_runtime_apis! {
         /// Extract the bundles from the given extrinsics.
         fn extract_bundles(extrinsics: Vec<Block::Extrinsic>) -> Vec<OpaqueBundle>;
 
+        /// Extract the receipts from the given extrinsics.
+        fn extract_receipts(
+            extrinsics: Vec<Block::Extrinsic>,
+        ) -> Vec<SignedExecutionReceipt<NumberFor<Block>, Block::Hash, SecondaryHash>>;
+
+        /// Extract the fraud proofs from the given extrinsics.
+        fn extract_fraud_proofs(extrinsics: Vec<Block::Extrinsic>) -> Vec<FraudProof>;
```

Comment:
There are two usages of `extract_fraud_proof`:

- https://github.com/subspace/subspace/blob/62576455159bcc813819331115af7fda8c3649b3/crates/sc-consensus-fraud-proof/src/lib.rs#L111
Replacing this usage with `extract_fraud_proofs` here is fine as we're operating upon the entire extrinsic list anyway.
- https://github.com/subspace/subspace/blob/62576455159bcc813819331115af7fda8c3649b3/crates/subspace-service/src/pool.rs#L148
The object of this usage is a single extrinsic, it can be replaced extract_fraud_proofs` as well but I'm not sure it should, no opinion on this.

Reply:
I do not have a strong opinion either, but I think reducing API surface would be nice. We probably can't just remove it right now, but we can add TODO to remove later when we do breaking API changes.

Reply:
TODO noted.